persistence/src/commonMain/db_user/com/wire/kalium/persistence/Users.sq

```diff
@@ -47,11 +47,12 @@ SELECT * FROM User WHERE connection_status = ?;
 selectByQualifiedId:
 SELECT * FROM User WHERE qualified_id = ?;
 
-selectByNameOrHandleOrEmail:
+selectByNameOrHandleOrEmailAndConnectionState:
 SELECT * FROM User
 WHERE name LIKE ('%' || :searchQuery || '%')
 OR  handle LIKE  ('%' || :searchQuery || '%')
-OR  email LIKE  ('%' || :searchQuery || '%');
+OR  email LIKE  ('%' || :searchQuery || '%')
+AND connection_status = :connectionStatus;
```

Comment:
I think first part with all the OR has to be in brackets otherwise you won't get what you want 😄

Reply:
hmmm unit tests are passing :P will verify it with another unit tests as Vitor suggested adding some that should give the answer to this one if it works as expected

Reply:
good one, could anyone explain me @vitorhugods @saleniuk  why we need those brackets ? 

does it work likes this without brackets ? : 

look for a ROW with a name equal to query 
OR a ROW with a handle equal to query 
OR a ROW equal to a EMAIL AND a connection_status equal to query ?

Reply:
AND has precedence over OR, so if we have something like that:
`WHERE A OR B AND C`
 without brackets it will be actually executed like
 `WHERE A OR (B AND C)`
 so if we want it to be different, we have to set the brackets ourself, in this case it should be:
 `WHERE (A OR B) AND C`

Reply:
> look for a ROW with a name equal to query
> OR a ROW with a handle equal to query
> OR a ROW equal to a EMAIL AND a connection_status equal to query ?

We want users that MUST be `CONNECTED`.

And match at least one of: `handle`, `email`, `name`.

Currently: `matchName || matchEmail || matchHandle && matchConnection`.
It means that if it `matchName` is true, it's enough.

It should be: `(matchName || matchEmail || matchHandle) && matchConnection`.



You can play with this example:
https://www.db-fiddle.com/f/dEjmVGeZqo5ZePLPzJukNv/0

Reply:
thanks <3